lib/rules/array-indentation.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @fileoverview Check indentation at the beginning and end of a literal array expression
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+   var sourceCode = context.getSourceCode(),
+       lines = sourceCode.getLines();
+
+   //--------------------------------------------------------------------------
+   // Helpers
+   //--------------------------------------------------------------------------
+
+   function lineIndent(lineNumber) {
+      var line = lines[lineNumber - 1];
+
+      if (line === undefined) {
+         throw new Error('Invalid lineNumber ' + lineNumber + ' supplied to lineIndent');
+      }
+
+      return line[0] === ' ' ? line.match(/[^ \t]/).index : 0;
```

Comment:
seems like your ternary condition only looks for spaces while the `match` looks for spaces or tabs